### PR TITLE
added enhancement #1703

### PR DESCRIPTION
**Pull request by mukulsinghsaini**
Make HOCON config ToString() method more readable
added enhancement #1703 
@Horusiath please review

### CODE REVIEW

**Comment by Horusiath** [comment]
For me it looks nice :) However I'd like to hear @rogeralsing if he sees any pitfalls, and @Aaronontheweb story about API approval process (since this one affect public API). Guys, are you ok with this one?

**Comment by Aaronontheweb** [comment]
@mukulsinghsaini could you please give this a try and re-submit your PR? http://getakka.net/docs/akka-developers/public-api-changes

Need to follow that process whenever we touch the public API.

File: src/core/Akka/Configuration/Config.cs [hunk 1]
```diff
@@ -393,6 +393,33 @@ public override string ToString()
         }
 
         /// <summary>
+        /// Converts the current configuration to a string 
+        /// </summary>
+        /// <param name="includeFallback">if true returns string with current config combined with fallback key-values else only current config key-values</param>
+        /// <returns></returns>
+        public string ToString(Boolean includeFallback)
```
Comment (rogeralsing) [inline src/core/Akka/Configuration/Config.cs]: I'd argue that the use of `Boolean` should be replaced with `bool`

**Comment by mukulsinghsaini** [comment]
hi..I would need a little help..on my system the following lines are generated
```
[System.Runtime.CompilerServices.IteratorStateMachineAttribute(typeof(Akka.Configuration.Config.<AsEnumerable>d__46))]
        public virtual System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, Akka.Configuration.Hocon.HoconValue>> AsEnumerable() { }

```
with only a difference of number d__46 instead of d__45, which I think is breaking the public api test.
What could be the reason ?

**Comment by mukulsinghsaini** [comment]
@Aaronontheweb CI tests are failing even after public api changes. Can you help me find if I am missing anything else ? Thanks.

File: src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt [hunk 2]
```diff
@@ -1968,6 +1968,7 @@ namespace Akka.Configuration
         public Akka.Configuration.Hocon.HoconValue GetValue(string path) { }
         public virtual bool HasPath(string path) { }
         public override string ToString() { }
+	public string ToString(bool includeFallback) { }
```
Comment (Silv3rcircl3) [inline src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt]: this line starts with a tab, you need to replace it with spaces

Comment (mukulsinghsaini) [inline src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt]: @Silv3rcircl3 tried it but still gives error !

File: src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt [hunk 3]
```diff
@@ -4902,4 +4903,4 @@ namespace Akka.Util.Reflection
     {
         public static System.Collections.Generic.IEnumerable<object> GetArguments(this System.Linq.Expressions.NewExpression newExpression) { }
     }
-}
\ No newline at end of file
+}
```
Comment (Silv3rcircl3) [inline src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt]: Could you please double check that there isn't another line after the breached? I have pulled you branch and that was the last issue I had with the approval test even if the line doesn't show up here.

By the way you know that you can run the test locally ? :smile: 
It's locaded in the Akka.API.Tests projects

Comment (mukulsinghsaini) [inline src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt]: @Silv3rcircl3 It did help thanks ! would be more careful next time.

Just one more thing...windows-multinode-tests are failing now...can a new public api break it possibly ?

sorry about my noobish questions !

Comment (sean-gilliam) [inline src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt]: @mukulsinghsaini multinode tests are known to be racy and fail often. it isn't related to your stuff so you're fine.

Comment (mukulsinghsaini) [inline src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt]: @sean-gilliam Thanks for the quick update. That was a relief ! :)

**Comment by Silv3rcircl3** [comment]
@mukulsinghsaini now you need to squash your commits into a single one ;-)

**Comment by mukulsinghsaini** [comment]
@Silv3rcircl3 I have squashed the commits.

**Comment by Silv3rcircl3** [comment]
@mukulsinghsaini unfortunately not all off them, when you look into the commits view you still see 4 commits including merges from upstream instead of one

**Comment by mukulsinghsaini** [comment]
@Silv3rcircl3 I see other people commits in-between my commits..so was cautious not to delete them accidently..plz see [https://github.com/mukulsinghsaini/akka.net/commits/dev](https://github.com/mukulsinghsaini/akka.net/commits/dev)

is it ok to squash them ?

**Comment by Silv3rcircl3** [comment]
@mukulsinghsaini good catch! I haven't noticed that you are working on direct dev branch instead of a separate feature branch, this makes things a little bit tricky.

@Aaronontheweb @rogeralsing would you be fine with merging these four commits or is there some git magic that is able to rebase inside a dev branch?

**Comment by mukulsinghsaini** [comment]
If its too complex, I can delete the current fork and raise a new pull request with new fork. Let me know if it helps.

**Comment by Silv3rcircl3** [comment]
@mukulsinghsaini that would be awesome :+1: The best way to create a PR is described in detail in [this](https://github.com/akkadotnet/akka.net/blob/dev/CONTRIBUTING.md) document, hope that helps.

**Pull request closed without merging**